ui/src/singletons/store.ts

```diff
@@ -299,6 +362,7 @@ export const useFlowStore = () =>
     useShallow((state: AppStore) => ({
       nodes: state.nodes,
       edges: state.edges,
+      layout: state.layout,
```

Comment:
I'm not sure that `layout` belongs in the `FlowStore`, since that it is very ReactFlow specific. You already created the `useGetLayout` hook, so you might as well use that to access the layout state.